src/common/WebLink.js

```diff
@@ -21,15 +21,52 @@ const componentStyles = createStyleSheet({
 
 /**
  * A button styled like a web link.
+ *
+ * Accepts `ZulipText`s, `ZulipTextIntl`s, and strings as children.
+ *
+ * Note: This sort of messes in those non-string children's business by
+ *   unsetting some of their default style attributes. It does this so that
+ *   its own special formatting can be passed down through the limited style
+ *   inheritance that RN supports:
+ *     https://reactnative.dev/docs/text#limited-style-inheritance
+ *   See `aggressiveDefaultStyle` in ZulipText.
+ *
+ * TODO: Possibly there's a better way handle this.
  */
 export default function WebLink(props: Props): Node {
+  const { children } = props;
+
   return (
-    <ZulipTextIntl
+    <ZulipText
       style={componentStyles.link}
-      text={props.label}
       onPress={() => {
         openLinkEmbedded(props.url.toString());
       }}
-    />
+    >
+      {React.Children.map(children, child => {
+        if (!React.isValidElement(child)) {
+          // E.g., a plain string, so not a ZulipText that could be trying
+          // to set its own styles. Our own ZulipText's styles will be
+          // applied directly.
+          return child;
```

Comment:
Huh, does this work -- are plain strings accepted as children here?

That name `isValidElement` seems to suggest that they wouldn't be.

Reply:
Here's the doc for `isValidElement`: https://reactjs.org/docs/react-api.html#isvalidelement

It's super short and lacks detail:

> `React.isValidElement(object)`
> 
> Verifies the object is a React element. Returns `true` or `false`.

My understanding (which the doc could do better to confirm or reject, like with a link to a definition) is that a "React element" is the interesting kind of node, typed as `React$Element`. The node that's an object with `type`, `props`, `key`, and `ref`. Your docs/background/react.md helped reinforce this understanding, and I think it's right. 🙂

`props.children` might be, or might contain, one of those. But IIUC it might also have other things, like `false` if in JSX you give a child like `isLoading && <LoadingSpinner />` and `isLoading` is false. It might have a string, as in `ZulipText` with a non-nullish `text`:

```js
      <Text style={[aggressiveDefaultStyle, style]} {...restProps}>
        {text}
        {children}
      </Text>
```

React Native has its _own_ rule against putting string nodes all over the place. Because they want the `Text` component to be the only thing that controls UI text formatting, React Native rejects element trees that have string nodes as direct children of non-`Text` elements. See the following bit of [doc](https://reactnative.dev/docs/text#limited-style-inheritance), which confusingly seems to call "string nodes" "text nodes" (but note the lowercase "t"):

> In React Native, we are more strict about it: **you must wrap all the text nodes inside of a `<Text>` component**. You cannot have a text node directly under a `<View>`.

But that constraint is managed by React Native, not React proper, so I wouldn't expect `React.isValidElement` to be aware of it.

Reply:
Perhaps I've missed something in your comment; have I answered what you were getting at? 🙂

Reply:
> My understanding (which the doc could do better to confirm or reject, like with a link to a definition) is that a "React element" is the interesting kind of node, typed as `React$Element`. The node that's an object with `type`, `props`, `key`, and `ref`. Your docs/background/react.md helped reinforce this understanding, and I think it's right. slightly_smiling_face

Ah, yeah, that makes sense.

From looking in `node_modules/react/`, and then in a React source tree for a cleaner version, here's the implementation:
```js
/**
 * Verifies the object is a ReactElement.
 * See https://reactjs.org/docs/react-api.html#isvalidelement
 * @param {?object} object
 * @return {boolean} True if `object` is a ReactElement.
 * @final
 */
export function isValidElement(object) {
  return (
    typeof object === 'object' &&
    object !== null &&
    object.$$typeof === REACT_ELEMENT_TYPE
  );
}
```

So that agrees -- it's an object, with the React marker property `$$typeof` set the way it is for React elements.

Reply:
Here's an attempt at rewording based on my understanding. Does this seem right?

```diff
         if (!React.isValidElement(child)) {
-          // E.g., a plain string, so not a ZulipText that could be trying
-          // to set its own styles. Our own ZulipText's styles will be
-          // applied directly.
+          // Some React node that isn't a React element (a `React.Element`);
+          // e.g., a plain string. The enclosing ZulipText will apply its
+          // styles directly.
```

Another bit reflected there is that I think "our own ZulipText" is meant to refer to the `ZulipText` element that this code is providing a child for, the one at the top level of the `WebLink` return value. I think when I previously read this PR I wasn't sure if it just meant the ZulipText component-type in general, with "our own" there to contrast it with something from upstream like `Text`.

I'll push to the branch with an extra couple of commits for that rewording; feel free to reorder and/or squash.